zetta_utils/tensor_ops/mask.py

```diff
@@ -287,3 +288,31 @@ def kornia_dilation(
     result_torch = result_torch.to(data_torch.dtype)
     result = convert.astype(einops.rearrange(result_torch, "Z C X Y -> C X Y Z"), data)
     return result
+
+
+@builder.register("mask_out_with_fn")  # type: ignore
+@supports_dict
+@skip_on_empty_data
+@typechecked
+def mask_out_with_fn(
+    data: TensorTypeVar, fn: Callable[[TensorTypeVar], TensorTypeVar]
+) -> TensorTypeVar:  # pragma: no cover # no logic
+    data_t = convert.to_torch(data)
+    mask_t = convert.to_torch(fn(data))
+    data_masked_t = torch.where(mask_t, torch.zeros_like(data_t), data_t)
+    result = convert.astype(data_masked_t, data)
+    return result
+
+
+@builder.register("combine_mask_fns")  # type: ignore
+@supports_dict
+@skip_on_empty_data
+@typechecked
+def combine_mask_fns(
+    data: TensorTypeVar, fns: Sequence[Callable[[TensorTypeVar], TensorTypeVar]]
+) -> TensorTypeVar:  # pragma: no cover # no logic
+    assert len(fns) > 0
+    masks_t = [convert.to_torch(fn(data)) for fn in fns]
+    result_t = reduce(torch.Tensor.add_, masks_t, torch.zeros_like(masks_t[0]))
```

Comment:
This should either be `Tensor.maximum_` or needs to handle potential overflow issues by widening the data type.